Modules/ITS/src/ITSRawTask.cxx

```diff
@@ -0,0 +1,771 @@
+    nBinsX = nChipsPerHic[aLayer];
+    hChipStaveOccupancy[aLayer] = new TH2D(Form("Occupancy/Layer%d/Layer%dChipStave", aLayer, aLayer),
+        Form("ITS Layer%d, Occupancy vs Chip and Stave", aLayer), nBinsX, -.5 , nBinsX-.5 , NStaves[aLayer], -.5, NStaves[aLayer]-.5);
+    formatAxes(hChipStaveOccupancy[aLayer], "Chip Number", "Stave Number", 1., 1.1);
+    formatStatistics(hChipStaveOccupancy[aLayer]);
+   // format2DZaxis(hChipStaveOccupancy[aLayer]);
+  }
+  else {
+    nBinsX = nHicPerStave[aLayer];
+    hChipStaveOccupancy[aLayer] = new TH2D(Form("Occupancy/Layer%d/Layer%dHicStave", aLayer, aLayer),
+        Form("ITS Layer%d, Occupancy vs Hic and Stave", aLayer), nBinsX, -.5 , nBinsX-.5 , NStaves[aLayer], -.5, NStaves[aLayer]-.5);
+    formatAxes(hChipStaveOccupancy[aLayer], "Hic Number", "Stave Number", 1., 1.1);
+    formatStatistics(hChipStaveOccupancy[aLayer]);
+   // format2DZaxis(hChipStaveOccupancy[aLayer]);
+  }
+
+  hChipStaveOccupancy[aLayer]->GetZaxis()->SetTitle("Number of Hits");
+  hChipStaveOccupancy[aLayer]->GetZaxis()->SetTitleOffset(1.4);
+  addObject(hChipStaveOccupancy[aLayer]);
+}
+
+// hEtaPhiHitmap: eta-phi hitmaps for complete layers, binning 100 x 100 pixels
+// using eta coverage of TDR, assuming phi runs from 0 ... 2*Pi
+void ITSRawTask::createEtaPhiHitmap(int aLayer)
+{
+  int NEta, NPhi;
+  if (aLayer < NLayerIB) {
+    NEta = 9 * 10;
+    NPhi = NStaves[aLayer] * 5;
+  }
+  else {
+    NEta = nHicPerStave[aLayer] * 70;
+    NPhi = NStaves[aLayer] * 10;
+  }
+  hEtaPhiHitmap[aLayer] = new TH2I(Form("Occupancy/Layer%d/Layer%dEtaPhi", aLayer, aLayer),
+      Form("ITS Layer%d, Hits vs Eta and Phi", aLayer), NEta, (-1)*etaCoverage[aLayer], etaCoverage[aLayer], NPhi, PhiMin, PhiMax);
+  formatAxes(hEtaPhiHitmap[aLayer], "#eta", "#phi", 1., 1.1);
+  hEtaPhiHitmap[aLayer]->GetZaxis()->SetTitle("Number of Hits");
+  hEtaPhiHitmap[aLayer]->GetZaxis()->SetTitleOffset(1.4);
+  addObject(hEtaPhiHitmap[aLayer]);
+}
+
+void ITSRawTask::createStaveHistos(int aLayer, int aStave)
+{
+  for (int iHic = 0; iHic < nHicPerStave[aLayer]; iHic++) {
+     createHicHistos(aLayer, aStave, iHic);
+  }
+}
+
+
+void ITSRawTask::createHicHistos(int aLayer, int aStave, int aHic)
+{
+  TString Name, Title;
+  int nBinsX, nBinsY, maxX, maxY, nChips;
+
+  if (aLayer < NLayerIB) {
+    Name = Form("Occupancy/Layer%d/Stave%d/Layer%dStave%dHITMAP", aLayer, aStave, aLayer, aStave);
+    Title = Form("Hits on Layer %d, Stave %d", aLayer, aStave);
+    maxX = 9 * NColHis;
+    maxY = NRowHis;
+nChips = 9;
+  }
+  else {
+    Name = Form("Occupancy/Layer%d/Stave%d/HIC%d/Layer%dStave%dHIC%dHITMAP", aLayer, aStave, aHic, aLayer, aStave, aHic);
+Title = Form("Hits on Layer %d, Stave %d, Hic %d", aLayer, aStave, aHic);
+    maxX = 7 * NColHis;
+    maxY = 2 * NRowHis;
+    nChips = 14;
+  }
+  nBinsX = maxX / SizeReduce;
+  nBinsY = maxY / SizeReduce;
+  hHicHitmap[aLayer][aStave][aHic] = new TH2I(Name, Title, nBinsX, 0, maxX, nBinsY, 0,  maxY);
```

Comment:
Could you make sure, that each dynamically allocated variable is later deleted during destruction of the task or other appropriate moment?

Reply:
Added to the deconstructor.